src/compiler/binder.ts

```diff
@@ -334,6 +344,9 @@ namespace ts {
 
             const isDefaultExport = hasModifier(node, ModifierFlags.Default);
 
+            if (isNamedDeclaration(node)) {

```

Comment:
If we were to add another condition to the `if`, what would that condition be?
What would be the advantage of not assigning `node.name.parent`?
(genuine question, I don't have a big view on how things fit together)

Reply:
In the current version of Typescript, the condition comes from stepping through the `if`s surrounding the current location: 

```js
name !== undefined 
&& symbol 
&& (!isReplaceableByMethod || symbol.isReplaceableByMethod) 
&& symbol.flags & excludes 
&& !symbol.isReplaceableByMethod 
&& !(includes & SymbolFlags.Variable && symbol.flags & SymbolFlags.Assignment)
```

I'm not sure which one of those is violated and caused the move to the beginning of the function. But I think it would be better to find out which tests fail, then craft a condition based on those tests and only set `node.name.parent` in cases where that condition is true.

I think there are efficiency advantages to not setting parent pointers, for space and perhaps for time. @rbuckton I think you have a clearer understanding of why we don't fill out parent nodes by default. What do you think about this?

Reply:
@sandersn We will eventually assign `.parent` anyways when we traverse the tree during bind, so setting it early shouldn't be an issue. That said, perhaps there's a way to remove the dependency on `.parent` from `getDeclarationName`.

Reply:
See https://github.com/microsoft/TypeScript/pull/30829/files#r300138764 for a possible solution.

Reply:
Thanks for explaining!

You were right that reinstating the original condition leads to an error when `getDeclarationName` calls `getContainingClass` (good :eyes:).

Looks like we can get away with not using node.parent

https://github.com/microsoft/TypeScript/pull/30829/files/a553660331caef8a396bc7e22c78730bc7ee3cad#discussion_r300138764